src/blockdev.rs

```diff
@@ -519,6 +519,13 @@ pub struct SavedPartitions {
     partitions: Vec<(u32, GPTPartitionEntry)>,
 }
 
+fn disk_has_mbr(disk: &mut File) -> bool {
```

Comment:
Nit: I'd probably move this next to `get_gpt_size` and make the argument signature match the one in that function.  Strictly speaking we don't need a `File`.

Reply:
Thats super cool, did not know I could do that!